lib/telemetry_metrics/console_reporter.ex

```diff
@@ -0,0 +1,146 @@
+defmodule Telemetry.Metrics.ConsoleReporter do
+  @moduledoc """
+  A reporter that prints events and metrics to the terminal.
+
+  This is useful for debugging and discovering all available
+  measurements and metadata in an event.
+
+  For example, imagine the given metrics:
+
+      metrics = [
+        last_value("vm.memory.binary", unit: :byte),
+        counter("vm.memory.total")
+      ]
+
+  A console reporter can be started as a child of your your supervision
+  tree as:
+
+      {Telemetry.Metrics.ConsoleReporter, metrics: metrics}
+
+  Now when the "vm.memory" telemetry event is dispatched, we will see
+  reports like this:
+
+      [Telemetry.Metrics.ConsoleReporter] Got new event!
+      Event name: vm.memory
+      All measurements: %{binary: 100, total: 200}
+      All metadata: %{}
+
+      Metric measurement: :binary (last_value)
+      With value: 100 byte
+      And tag values: %{}
+
+      Metric measurement: :total (counter)
+      With value: 200
```

Comment:
I'm wondering if we shouldn't always print `1` as measurement value for counters. Or maybe add an info that counter's measurement is not relevant.

Reply:
Even better: I won't show the value for them. Good catch!